tests/acl/test_src_mac_rewrite.py

```diff
@@ -0,0 +1,705 @@
+    remove_acl_rules(duthost)
+
+
+def _send_and_verify_mac_rewrite(ptfadapter, ptf_port_1, ptf_port_2_list, duthost,
+                                 src_ip, dst_ip, orig_src_mac, expected_inner_mac,
+                                 vni_id, outer_src_mac, outer_dst_mac, outer_src_ip, outer_dst_ip,
+                                 table_name, rule_name):
+    """
+    Send a packet and verify MAC rewrite with VXLAN encapsulation.
+    Updated to follow the pattern from test_bgp_vnet_route_forwarding.
+
+    Args:
+        ptf_port_2_list: List of PTF ports where packet might be received (like [24,25,26,27])
+    """
+    router_mac = duthost.facts["router_mac"]
+
+    # Create input packet
+    options = {'ip_ecn': 0}
+    pkt_opts = {
+        "pktlen": 100,
+        "eth_dst": router_mac,
+        "eth_src": orig_src_mac,
+        "ip_dst": dst_ip,
+        "ip_src": src_ip,
+        "ip_id": 105,
+        "ip_ttl": 64,
+        "tcp_sport": 1234,
+        "tcp_dport": 5000
+    }
+
+    pkt_opts.update(options)
+    input_pkt = testutils.simple_tcp_packet(**pkt_opts)
+
+    # Create expected inner packet with rewritten MAC
+    inner_pkt_opts = {
+        "pktlen": 100,
+        "eth_dst": "00:12:34:56:78:9a",  # Switch MAC
+        "eth_src": expected_inner_mac,  # This should be the rewritten MAC
+        "ip_dst": dst_ip,
+        "ip_src": src_ip,
+        "ip_id": 105,
+        "ip_ttl": 63,  # TTL decremented
+        "tcp_sport": 1234,
+        "tcp_dport": 5000
+    }
+    inner_pkt_opts.update(options)
+    expected_inner_pkt = testutils.simple_tcp_packet(**inner_pkt_opts)
+
+    # Create expected VXLAN encapsulated packet
+    encap_pkt = testutils.simple_vxlan_packet(
+        eth_src=router_mac,
+        eth_dst="00:aa:bb:cc:dd:ee",  # Random MAC
+        ip_id=0,
+        ip_ihl=5,
+        ip_src="10.1.0.32",  # DUT VTEP
+        ip_dst="100.0.1.10",  # PTF VTEP from config
+        ip_ttl=128,
+        udp_sport=49366,
+        udp_dport=4789,  # Standard VXLAN port
+        with_udp_chksum=False,
+        vxlan_vni=10000,  # VNI from config
+        inner_frame=expected_inner_pkt,
+        **options
+    )
+
+    # Set IP flags
+    encap_pkt[IP].flags = 0x2
+
+    # Create masked expected packet with detailed masking
+    masked_exp_pkt = mask.Mask(encap_pkt)
+    masked_exp_pkt.set_ignore_extra_bytes()
+
+    # Outer headers masking - mask everything we don't care about
+    masked_exp_pkt.set_do_not_care_scapy(Ether, "src")
+    masked_exp_pkt.set_do_not_care_scapy(Ether, "dst")
+    masked_exp_pkt.set_do_not_care_scapy(IP, "ttl")
+    masked_exp_pkt.set_do_not_care_scapy(IP, "chksum")
+    masked_exp_pkt.set_do_not_care_scapy(IP, "id")
+    masked_exp_pkt.set_do_not_care_scapy(IP, "src")
+    masked_exp_pkt.set_do_not_care_scapy(IP, "dst")
+    masked_exp_pkt.set_do_not_care_scapy(UDP, "sport")
+    masked_exp_pkt.set_do_not_care_scapy(UDP, "chksum")
```

Comment:
not required? I think you are not doing udp_chksum as per the packet generated in exp packet